### DISCUSSION TITLE
Come aggiungere regime fiscale e TipoDocumento per autofatture con Zapier

### DISCUSSION THREAD

**Discussion by DifferentSailing** [fattureincloud/api#410] [Q&A] [answered]
Ciao a tutti, 

Sto cercando di creare con Zapier delle autofatture e devo impostare i campi Regime Fiscale (sempre rf18) e Tipologia documento (td16 o td17). é possibile farlo? se si come?

@emajo magari hai qualche idea? 

Grazie in anticipo!! 
Federico.

**Comment by DifferentSailing** [comment]
Per tipo Documento ho provato ad inserire questo JSON nel campo Issued document advanced raw attributes for e-invoices - [data.items_list.ei_raw] ma senza successo: 

{
    "data": {
        "documents_type_info": {
            "TD16": {
                "self_own_invoice": false,
                "self_supplier_invoice": true,
                "description": "TD16 - integrazione fattura reverse charge interno"
            }
        }
    }
}

Credo sia il modo giusto, ma non sta funzionando.

**Comment by emajo** [comment]
Ciao @DifferentSailing, il campo corretto da utilizzare è `data.ei_raw` e va valorizzato così:
```
{
        "ei_raw": {
            "FatturaElettronicaBody": {
                "DatiGenerali": {
                    "DatiGeneraliDocumento": {
                        "TipoDocumento": "TD16"
                    }
                }
            }
      }
}
```
grazie,
Emanuele.

**Reply by DifferentSailing** [reply]
Grazie! Purtroppo anche con questo setup non funziona, nella fattura elettronica ancora vedo TD01. 
<img width="810" alt="Screenshot 2024-02-14 154725" src="https://github.com/fattureincloud/api/assets/159318956/4ab53535-b8df-4b2d-830b-49aa578db75f">


In più avrei anche bisogno di modificare i campi: 
- Regime fiscale
- Data fattura collegata
- Numero fattura collegata

é fattibile?

**Reply by emajo** [reply, marked as answer]
Ciao, effettivamente l'esempio precedente non era corretto, così dovrebbe coprire tutte le casistiche descritte:

```
{
            "FatturaElettronicaBody": {
                "DatiGenerali": {
                    "DatiGeneraliDocumento": {
                        "TipoDocumento": "TD17"
                    },
                    "DatiFattureCollegate": [
                        {
                            "IdDocumento": "123456789",
                            "Data": "2024-02-14"
                        }
                    ]
                }
            },
            "FatturaElettronicaHeader": {
                "CedentePrestatore": {
                    "DatiAnagrafici": {
                        "RegimeFiscale": "RF02"
                    }
                }
            }
        }
```

**Reply by DifferentSailing** [reply]
Grazie @emajo , funziona!!
